src/main/java/org/scijava/ui/swing/widget/SwingNumberWidget.java

```diff
@@ -132,6 +137,17 @@ else if (model.isStyle(NumberWidget.SLIDER_STYLE)) {
 		syncSliders();
 	}
 
+	private String getFormat(WidgetModel model) {
+		final String widgetStyle = model.getItem().getWidgetStyle();
+		String format = null;
+		for (final String s : widgetStyle.split(",")) {
+			if (s.startsWith("format:")) {
+				format = s.substring(s.indexOf(":") + 1).trim();
+			}
+		}
```

Comment:
With https://github.com/scijava/scijava-common/pull/384, I tried to centralize parsing of the `style` attribute into a `WidgetStyle` class (as proposed by @ctrueden) and to fix some related issues, such as inconsistent handling of spaces with style definitions.

To keep the code base as DRY as possible, I suggest relying on that `WidgetStyle` class instead of this `getFormat` implementation. The latter would be specific to `scijava-ui-swing` whereas the former would allow usage from any UI implementation.

Reply:
@imagejan , basically that does sound like a good idea, but that PR has been open since April. We would rather have this be merged and available sometime soon and when your widget style becomes available, start using it as of then. 

We are however not sure what the release cycle is. Is there a chance we can get this code released earlier if we keep it contained to just ui-swing?

Reply:
I merged scijava/scijava-common#384 and released scijava-common 2.85.0. @sunsear @BoudewijnvanLangerak Could you please add `<scijava-common.version>2.85.0</scijava-common.version>` to the POM on this branch, and try to implement @imagejan's suggestions? I'll then try to get this PR merged quickly. I'm sorry I'm so slow merging PRs—I maintain too many things, and so the release cycle of everything suffers.

Reply:
With https://github.com/scijava/scijava-common/pull/405, it should be possible to replace this `getFormat()` method by a call to:

```java
WidgetStyle.getStyleModifier(model.getItem().getWidgetStyle(), "format")
```